### ISSUE TITLE
Difference between original parents and original leaves in search view 

### ISSUE THREAD

**Issue by rasoulabbasi20n** [vakata/jstree#1562]
Hi, in search scenario, I want to remove checkbox icon if the node is a leaf in original-view (generated DOM for tree without any search criteria ) in the search-view. I read about how to remove checkbox for parents in original-view like this:

`#jstree .jstree-open > .jstree-anchor > .jstree-checkbox ,
#jstree .jstree-closed > .jstree-anchor > .jstree-checkbox {
 display: none !important;
}`

In my case, the DOM is specified by "jstree-search" class, but I don't know how to separate leaves from parents.

`#jstree .jstree-leaf> .jstree-search > .jstree-checkbox {
 display: none !important;
}`

"jstree-leaf " class is used for both original parents and leaves in search-view. 
In other words, in search-view, leaves include both parents and leaves from original-view.

Thanks in advance

**Comment by vakata** [comment]
Basically - you can't - leaves are part of the visualization. You have two options:
1) enable the [show_only_matches_children option](https://www.jstree.com/api/#/?q=search.show&f=$.jstree.defaults.search.show_only_matches_children) - that way your approach will work, but I am not sure this is a desired effect.
2) make a custom class to apply to all leave nodes - either passing that class along with the data (using the `a_attr` property), or as nodes are loaded (on `model.jstree` traverse all nodes and apply the class).

Tell me if you need any more help with any of the above options.
